internal/provider/data_source_packer_image_test.go

```diff
@@ -134,3 +155,62 @@ func TestAcc_dataSourcePackerImage_revokedIteration(t *testing.T) {
 		},
 	})
 }
+
+func TestAcc_dataSourcePackerImage_channelAndIterationIDReject(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			// basically just testing that we don't pass validation here
+			{
+				PlanOnly:    true,
+				Config:      testConfig(testAccPackerImageBothChanAndIter),
+				ExpectError: regexp.MustCompile("Error: Invalid combination of arguments"),
+			},
+		},
+	})
+}
+
+func TestAcc_dataSourcePackerImage_channelAccept(t *testing.T) {
+	fingerprint := "acceptChannel"
+	resourceName := "data.hcp_packer_image.arch-btw"
+	var buildID string
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t, map[string]bool{"aws": false, "azure": false}) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				PlanOnly: true,
+				PreConfig: func() {
+					// CheckDestroy doesn't get called when the test fails and doesn't
+					// produce any tf state. In this case we destroy any existing resource
+					// before creating them.
```

Comment:
You're right, I did paste this from the failing test without too much thought. The other test uses `CheckDestroy` already, so that's not a big problem I imagine.
Though semantically, is this the right way to do this? My understanding was that CheckDestroy was only to test that resources are indeed removed, not to do the action of removing.
I imagine that would be done as an extra step for the teardown?